finetune/base.py

```diff
@@ -54,6 +38,8 @@ def __init__(self, config=None, **kwargs):
         :param config: A config object generated by `finetune.config.get_config` or None (for default config).
         :param **kwargs: key-value pairs of config items to override.
         """
+
+        atexit.register(self.__del__)
```

Comment:
I'm like about 90% certain that this already happens / is redundant with normal python behavior?

Reply:
it doesnt, CPython does not guarantee deconstructing all objects. Especially with test failures, I was seeing hundreds of GB of temp files

Reply:
Curious -- good to know.  Thanks for the info!